src/linux/wifi/core/AccessPointControllerLinux.cxx

```diff
@@ -197,7 +285,72 @@ AccessPointControllerLinux::SetProtocol(Microsoft::Net::Wifi::Ieee80211Protocol
     }
 
     // Reload hostapd conf file.
-    return isOk && m_hostapd.Reload();
+    isOk = isOk && m_hostapd.Reload();
+
+    return isOk;
+}
+
+bool
+AccessPointControllerLinux::SetAkmSuites(std::vector<Ieee80211AkmSuite> akmSuites)
+{
+    bool isOk = false;
+
+    // Construct the wpa_key_mgmt hostapd property value.
+    std::string concatenatedWpaKeyMgmtPropertyValue{};
+    for (const auto& akmSuite : akmSuites) {
+        const auto& wpaKeyMgmtPropertyValue = detail::IeeeAkmSuiteToWpaKeyMgmtPropertyValue(akmSuite);
+        concatenatedWpaKeyMgmtPropertyValue += wpaKeyMgmtPropertyValue;
+        concatenatedWpaKeyMgmtPropertyValue += " ";
+    }
+
+    try {
+        // Set the hostapd wpa_key_mgmt property.
+        isOk = m_hostapd.SetProperty(Wpa::ProtocolHostapd::PropertyNameWpaKeyMgmt, concatenatedWpaKeyMgmtPropertyValue);
+
+        // For simplicity, set the hostapd wpa property to allow WPA and WPA2 (and WPA3).
+        isOk = isOk && m_hostapd.SetProperty(Wpa::ProtocolHostapd::PropertyNameWpa, Wpa::ProtocolHostapd::PropertyWpaValueWpaAndWpa2);
+
+        // For simplicity, set the hostapd auth_algs property to allow both Open System Authentication and Shared Key Authentication.
+        isOk = isOk && m_hostapd.SetProperty(Wpa::ProtocolHostapd::PropertyNameAuthAlgs, Wpa::ProtocolHostapd::PropertyAuthAlgsValueOpenAndSharedKey);
```

Comment:
Is the intention to do something different in the future, or, just leave this as-is? If the former, let's add a comment.

Reply:
I was thinking of leaving it as-is, but I suppose there could be a situation where, for example, a client only wants to use the original WPA. Currently, with this set the way it's set now, hostapd will likely choose the best WPA version that it can since all are enabled